### ISSUE TITLE
Terminal : Control request/response

### ISSUE THREAD

**Issue by coldhack** [primefaces/primefaces#6217]
Having the possibility to define (i.e., add) output from beans would increase the utility for Terminal component (e.g.: allowing the execution of interactive commands ou pseudo-commands).

Example scenario:
Shell access to remote VM(Virtual machine) through JSF application using Terminal component.

**Comment by melloware** [comment]
Hmmm can you give a little more concrete example?  I am not sure what exactly you mean.   Can you even give an example terminal conversation?

In this example: https://www.primefaces.org/showcase/ui/misc/terminal/basic.xhtml

Can't you basically do whatever you want in `commandHandler="#{terminalBasicView.handleCommand}"` and process Shell commands?

**Comment by coldhack** [comment]
Imagine that you execute the `top` command on shell, there's a possible way to show successive outputs? Or you try to install using `yum` and it requires confirmation for some packages, as I can see it is impossible to manage, I think. Our `tail -f`, etc...

As it is, we have something to "request-response model", i.e., the `p:terminal` expects simples outputs from commands, and we can't control or manipulate the outputs presented on terminal, nor the command execution (the ability to receive new commands, or not).

**Comment by melloware** [comment]
Ahhh I understand now, you need something more than just a `String` result to make it more of an interactive console.

**Comment by coldhack** [comment]
Exactly!! Have more control on it. I've tried through binding, but no success.

**Comment by melloware** [comment]
@coldhack 5 years and no other upvotes and this component is not heavily used as the same thing can be done with simple websocket.  Going to close this for now
